### PR TITLE
Instant Search: separate out js build process.

### PR DESCRIPTION
**Pull request by gibrown**
In order to merge onto wp.com we need a separate build process. See: pMz3w-aAY-p2

I am uncertain if this is completely correct, but it still seems to work.

### CODE REVIEW

**Review by jeherve — COMMENTED** [review]
I think it may be worth updating `modules/search/instant-search/README.md` to mention how one can work with the module.
Something like what's done in https://github.com/Automattic/jetpack/blob/b7229b4ef78c4f8a389b340a7e25debfc20ee5bb/extensions/README.md#L147 for example.

cc @sirreal @ockham @simison who have some experience with the build process for blocks and using it on WordPress.com.

File: yarn.lock
Comment (sirreal) [inline yarn.lock]: lockfile changes look erroneous in this PR. I don't think there should be any, and you may be using an outdated version of yarn.

Comment (gibrown) [reply yarn.lock]: Removed. i'm actually on a newer version I guess. I'm not really sure why the integrity checks are failing, but fairly convinced they don't matter.

**Comment by gibrown** [comment]
I think I've addressed all the feedback.

**Review by jeherve — APPROVED** [review]
This looks good to me, and seems to work. I only have one question below.

I'll let others review and test as well. 👍

File: webpack.config.search.js
Comment (gibrown) [reply webpack.config.search.js]: Huh, I guess not. Good point.

Let's get this piece merged and then deal with that in a separate PR though. I may need to go back and forth to figure out how to get the paths and build process to work for both.

**Comment by gibrown** [comment]
@jeherve are you able to merge this? I am blocked by the "duplicate code" checks, but it seems like we want them to be duplicate due to the sync to wp.com

**Review by jeherve — APPROVED** [review]
Let's give this a try!

**Comment by jeherve** [comment]
@gibrown @jsnmoon Now that this has been merged, it may be worth to rebase the `instant-search-master` branch against `master` to test this in your next iterations.

I'll also leave the wpcom side of this PR to you, depending on how you want to proceed with the automation there.

**Pull request merged**
